lightning/src/ln/channelmanager.rs

```diff
@@ -2257,27 +2324,19 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 			// we got all the HTLCs and then a channel closed while we were waiting for the user to
 			// provide the preimage, so worrying too much about the optimal handling isn't worth
 			// it.
-
-			let (is_mpp, mut valid_mpp) = if let &Some(ref data) = &sources[0].payment_data {
-				assert!(payment_secret.is_some());
-				(true, data.total_msat >= expected_amount)
-			} else {
-				assert!(payment_secret.is_none());
-				(false, false)
-			};
-
+			let mut valid_mpp = true;
 			for htlc in sources.iter() {
-				if !is_mpp || !valid_mpp { break; }
 				if let None = channel_state.as_ref().unwrap().short_to_id.get(&htlc.prev_hop.short_channel_id) {
 					valid_mpp = false;
+					break;
 				}
 			}
 
 			let mut errs = Vec::new();
 			let mut claimed_any_htlcs = false;
 			for htlc in sources.drain(..) {
-				if channel_state.is_none() { channel_state = Some(self.channel_state.lock().unwrap()); }
-				if (is_mpp && !valid_mpp) || (!is_mpp && (htlc.value < expected_amount || htlc.value > expected_amount * 2)) {
+				if !valid_mpp {
+					if channel_state.is_none() { channel_state = Some(self.channel_state.lock().unwrap()); }
```

Comment:
Don't think this check: `htlc.value > expected_amount * 2` was migrated anywhere, is it needed?

Reply:
It only really applies for non-MPP - pre-MPP "don't overpay massively" was one (kinda poor) heuristic suggested in the spec to ignore sends from people other than the invoice-recipient. Once we have the payment secret, we don't really care too much, and the sender (who proved their knowledge of the payment preimage) sends also the specific amount we should expect, which we enforce is identical.